packages/as-types/assembly/serialization/arrays.ts

```diff
@@ -13,39 +20,34 @@ import { Serializable } from '../serializable';
  * @see {@link Serializable}
  *
  * @param source - the array to convert
+ * @returns The converted StaticArray<u8> (byte array) representation of the native type array.
  */
 export function nativeTypeArrayToBytes<T>(source: T[]): StaticArray<u8> {
   const sourceLength = source.length;
 
-  // ensures that the new array has the proper length.
-  // u16 are encoded using 4 bytes, u64 uses 8
-  // hence we need to multiple by 2 (swap bit) the right number of time (alignof<T>)
   let targetLength = (<usize>sourceLength) << alignof<T>();
 
-  // allocates a new StaticArray<u8> in the memory
   let target = changetype<StaticArray<u8>>(
     // @ts-ignore: Cannot find name '__new'
     __new(targetLength, idof<StaticArray<u8>>()),
   );
 
-  // copies the content of the source buffer to the newly allocated array.
-  // Note: the pointer to the data buffer for Typed Array is in dataStart.
-  // There is no such things for StaticArray.
```

Comment:
I wouldn't have deleted this.
What do you think @Ben-Rey ?

Reply:
I agree, it is useful for the understanding

Reply:
Maybe we can find a way to put what is important in a @PrivateRemarks

Reply:
Not so good in privateRemarks. 

```
/**
 * Convert an array of type parameter to StaticArray<u8>
 * 
 * @remarks
 * This will perform a deep copy only for native types.
 * It is inspired by https://github.com/AssemblyScript/assemblyscript/blob/main/std/assembly/array.ts#L69-L81.
 *
 * @privateRemarks
 * The length of the new array is calculated based on the length of the source array and the alignment of the type parameter T.
 * A new StaticArray<u8> is allocated in memory and the content of the source buffer is copied to the newly allocated array.
 * Note: the pointer to the data buffer for Typed Array is in dataStart.
 * There is no such thing for StaticArray.
 *
 * @param source - the array to convert
 * 
 * @returns the converted StaticArray<u8>
 */
export function nativeTypeArrayToBytes<T>(source: T[]): StaticArray<u8> {
  const sourceLength = source.length;

  let targetLength = (<usize>sourceLength) << alignof<T>();

  let target = changetype<StaticArray<u8>>(__new(targetLength, idof<StaticArray<u8>>()));

  memory.copy(changetype<usize>(target), source.dataStart, targetLength);

  return target;
}
```

Reply:
I think we can leave it for now

Reply:
> I think we can leave it for now

Should I includethem back in or leave it as is ?

Reply:
Yes include them back please

Reply:
> Yes include them back please

Added them as privateRemarks